crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -291,8 +291,13 @@ def disable_non_core_proposals
 
   def prepare_nodes_for_os_upgrade
     upgrade_nodes = NodeObject.all.reject { |node| node.admin? || node[:platform] == "windows" }
+    admin_node = NodeObject.admin_node
+    admin_node["target_platform"] = admin_node["provisioner"]["default_os"]
+    admin_node.save
```

Comment:
Just for the admin node. An option is to just unset it during the restore; there's code elsewhere that will automatically reset it to the right value, I think.